Indexing/SubstitutionTree.hpp

```diff
@@ -86,7 +85,7 @@ class InstanceCntr {
   Cntr& _cntr;
 
   InstanceCntr& operator=(InstanceCntr&& other) 
-  { swap(other._cntr, _cntr); return *this; }
+  { std::swap(other._cntr, _cntr); return *this; }
```

Comment:
Could you convert this and other occurrences to `using std::swap; swap(...)` please? Let me know if you need an explanation, I personally find this extremely unintuitive.

Reply:
Shall I put `using std::swap;` on top with other `using ` statements?

Reply:
I would suggest to do it only locally, since we are talking about headers!

Reply:
Also, keeping this together highlights it's an idiom and not something overlooked!